docs/serving/spec/knative-api-specification-1.0.md

```diff
@@ -1376,22 +1434,17 @@ Restrictions to the values of the field are noted in the Description column.
    </td>
    <td>int
 <br>
-(Required)
+(Optional)
 <br>
 Min: 0
 <br>
 Max: 100
    </td>
-   <td>The
-
-<a href="#request-routing">percentage of requests which should be allocated from
-the main Route domain name</a> to the specified <code>revisionName</code> or
-<code>configurationName</code>.
-
+   <td>The <code>percent</code> is optionally used to specify the percentage of requests which should be allocated from the main Route domain name to the specified <code>revisionName</code> or <code>configurationName</code>.
 <p>
-All <code>percent</code> values in <code>traffic</code> MUST sum to 100.
+To indicate that percetage based routing is to be used, all <code>traffic</code> sections for a single Service MUST have a <code>percent</code> value, and all values MUST sum to 100.
```

Comment:
```suggestion
To indicate that percentage based routing is to be used, all <code>traffic</code> sections for a single Service MUST have a <code>percent</code> value, and all values MUST sum to 100.
```

Hmm... I don't think requiring all sections to have a `percent` value is what we want (see discussion I mentioned above). Today we allow omitting the percentage value when it is zero, like so:

```yaml
  traffic:
    - revisionName: xyz-123
      percent: 100
    - revisionName: xyz-124
      tag: canary
```

I think you want to say:

> If any target within a `traffic` block has a non-zero `percent` value, this indicates that percentage-based routing is to be used. In this case, the the `percent` values MUST sum to 100.

(or the equivalent)

Reply:
@evankanderson ok I modified the text - see what you think